cms/djangoapps/contentstore/utils.py

```diff
@@ -296,7 +296,7 @@ def get_group_display_name(user_partitions, xblock_display_name):
     """
     for user_partition in user_partitions:
         for group in user_partition['groups']:
-            if str(group['id']) in xblock_display_name:
+            if _(u'Group ID {group_id}').format(group_id=group['id']) == xblock_display_name:
```

Comment:
This is better, but there are still issues with the general move functionality with regard to split_test instances. I documented those in https://openedx.atlassian.net/browse/TNL-6754.

It's also theoretically possible that someone could create a unit/vertical/sequential with "Group ID XXXX", and you would improperly replace it, but it does some like a very remote chance. Though remember that XXXX may be something just like 0, as in "Group ID 0" or "Group ID 1" (as you are iterating through all user partitions). You could further reduce the likelihood of incorrect matches by only looking at user_partitions that are related to the scheme "random" (RandomUserPartitionScheme), which is what the split_test module uses.